docs/security/clerk-csp.mdx

```diff
@@ -54,5 +54,107 @@ module.exports = nextConfig;
 ## Usage of `unsafe-eval` and `unsafe-inline` directives in Next.js
 
 - Within `script-src`, `unsafe-eval` is a [requirement for Next.js](https://nextjs.org/docs/app/building-your-application/configuring/content-security-policy#without-nonces) to run in development mode. _For production environment CSPs, it should be removed_.
-- Within `script-src`, `unsafe-inline` is a requirement for Next.js in both dev and prod environments if you're using the App Router. If you are using the Pages Router, it should be removed.
+- Within `script-src`, `unsafe-inline` is a requirement for Next.js in both dev and prod environments if you're using the App Router and not using `strict-dynamic`. If you are using the Pages Router, it can be removed.
 - Within `style-src`, `unsafe-inline` is a requirement for Clerk's components to inject their styles. Removing this requirement is on our roadmap - please [reach out to support](/support){{ target: '_blank' }} if you'd like to see this implemented sooner.
+
+## Implementing a `strict-dynamic` CSP
+
+If you'd like to implement a [strict-dynamic CSP](https://content-security-policy.com/strict-dynamic/), Clerk supports this, but with a different type of configuration. As strict-dynamic CSPs require a "nonce" value that should be programmatically generated per-request, the best way to make this work is within middleware that runs on every request. The following example demonstrates how to implement a strict-dynamic CSP with Next.js middleware, but the same approach could be used with any other framework.
+
+```js {{ prettier: false, filename: 'middleware.ts' }}
+  import { NextResponse } from "next/server";
+  import { clerkMiddleware } from "@clerk/nextjs/server";
+
+  export default clerkMiddleware((auth, request) => {
+    return applyCsp(request);
+  });
+
+  function applyCsp(request) {
+    // create a randomly generated nonce value
+    const nonce = Buffer.from(crypto.randomUUID()).toString("base64");
+
+    // format the CSP header
+    const cspHeader = `
```

Comment:
The CSP directives here differ from the one earlier in the page, let's make sure they're consistent (considering the usage of `strict-dynamic`)

Reply:
@jescalan this is missing `frame-src`